docs/GettingStartedDocs/Contributors/SignUsingEngines.md

```diff
@@ -0,0 +1,113 @@
+# Signing a Build With OpenSSL Engines
+
+OpenSSL allows extensions called "engines" to perform the generation of keys, signing and encryption.
+Engines are useful. They can be used to allow hardware offload of encryption duties, or improved
+encapsulation of secrets. Use of engines is required in some build systems as a way to avoid the
+storage of keys in plain text .pem files.
+
+oesign has been equipped with the ability to specify engines in the enclave signing process. In order
+to use the engine, the command line syntax has been extended.
+```
+oesign sign --enclave-image ENCLAVE_IMAGE --config-file CONFIG_FILE --engine engine-NAME -load-path engine-LOAD-PATH-key KEY-ID
+
+```
+
+## Engines
+
+The concept of an engine was added to OpenSSL starting with 0.9.6. There are built in engine implementations
+implementations for
+- Microsoft CryptoAPI
+- VIA Padlock
+- nCipher CHIL
+and a API for controlling dynamically bound engines which are bound from shared libraries at runtime via either program logic
```

Comment:
nit: still need to change "a API" to "an API"

Reply:
Shades of dickens!